controllers/PostsService.js

```diff
@@ -38,6 +39,41 @@ const addUserData = async (user, post) => {
   }, post);
 };
 
+const getDefaultPosts = async ({ limit, page, query, sort }) => {
+  let postLimit = Math.ceil(limit / 2);
+  // get sources
+  const sources = await Source.list({
+    query: {
+      isDeleted: false,
+    },
+    select: '_id',
+  });
+
+  const promises = sources.map(async ({ _id }) => {
```

Comment:
add index to map func call `const promises = sources.map(async ({ _id }, index) => {`